test/new-e2e/examples/agentenv_logs_test.go

```diff
@@ -7,41 +7,49 @@ package examples
 
 import (
 	_ "embed"
-	"time"
-
 	"testing"
+	"time"
 
 	fi "github.com/DataDog/datadog-agent/test/fakeintake/client"
-	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/utils/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/e2e"
+	"github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments"
+	awshost "github.com/DataDog/datadog-agent/test/new-e2e/pkg/environments/aws/host"
+
 	"github.com/DataDog/test-infra-definitions/components/datadog/agentparams"
+
 	"github.com/stretchr/testify/assert"
 )
 
 type vmFakeintakeSuite struct {
-	e2e.Suite[e2e.FakeIntakeEnv]
+	e2e.BaseSuite[environments.Host]
 }
 
 //go:embed testfixtures/custom_logs.yaml
 var customLogsConfig string
 
 func TestE2EVMFakeintakeSuite(t *testing.T) {
-	e2e.Run(t, &vmFakeintakeSuite{}, e2e.FakeIntakeStackDefWithDefaultVMAndAgentClient(
-		agentparams.WithIntegration("custom_logs.d", customLogsConfig), agentparams.WithLogs()),
-	)
+	e2e.Run(t, &vmFakeintakeSuite{}, e2e.WithProvisioner(
+		awshost.Provisioner(
+			awshost.WithAgentOptions(
+				agentparams.WithIntegration("custom_logs.d", customLogsConfig),
+				agentparams.WithLogs(),
+			),
+		),
+	))
 }
 
 func (s *vmFakeintakeSuite) TestLogs() {
-	fakeintake := s.Env().Fakeintake
+	fakeintake := s.Env().FakeIntake.Client()
 	// part 1: no logs
 	s.EventuallyWithT(func(c *assert.CollectT) {
 		logs, err := fakeintake.FilterLogs("custom_logs")
 		assert.NoError(c, err)
 		assert.Equal(c, len(logs), 0, "logs received while none expected")
 	}, 5*time.Minute, 10*time.Second)
-	// part 2: generate logs
-	s.Env().VM.Execute("echo 'totoro' > /tmp/test.log")
-	// part 3: there should be logs
 	s.EventuallyWithT(func(c *assert.CollectT) {
+		// part 2: generate logs
+		s.Env().RemoteHost.MustExecute("echo 'totoro' >> /tmp/test.log")
```

Comment:
<!-- Questions are appropriate if you have a potential concern but are not quite sure if it's relevant or not. Asking the author for clarification or investigation can lead to a quick resolution. -->
**❓ question**
Why is this moved inside the loop ? It is going to generate n logs, while we want to generate once and wait for the log to reach the fakeintake

Reply:
Because the test was flaky, I did not fully investigate but as it is an example, I prefer to have it working always.